lib/use_cases/update_rent_charges.rb

```diff
@@ -0,0 +1,99 @@
+class UpdateRentCharges
+  def initialize(fixed_data_gateway:, rent_charges_gateway:)
+    @fixed_data_gateway = fixed_data_gateway
+    @rent_charges_gateway = rent_charges_gateway
+  end
+
+  def execute(fixed_data)
+    errors = validations(fixed_data)
+
+    return { successful: false, errors: errors } unless errors.empty?
+  
+    rent_charges_for_given_year = @rent_charges_gateway.all((fixed_data[:year].to_i) - 1)
+    @rent_charges_for_given_year = rent_charges_for_given_year.map do |rent_charge|
+      Domain::RentCharge.new(
```

Comment:
I wouldn't construct the Domain::RentCharge object here, for a couple of reasons:

a) It's only used within this file currently, where it is immediately mapped into another hash by `rent_charges`, so I don't think it's providing a lot of value?
b) The main benefit of using Domain Objects comes when returning them from gateways. It allows all gateways which adhere to a responsibility/interface (e.g. SqlRentChargeGateway, ApiRentChargeGateway, InMemoryRentChargeGateway) to return a standardised object with behaviour that's useful to the use case that receives them. 

I'd suggest constructing the Domain::RentCharge object in the RentChargesGateway and returning it from the `#all` method, rather than using it in use case.

Correct me if I'm missing anything!

Reply:
yep sounds good!